test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmark
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	testutils "k8s.io/kubernetes/test/utils"
+	ktesting "k8s.io/kubernetes/test/utils/ktesting"
+)
+
+// TestRunCreateNodesOp verifies that runCreateNodesOp creates the
+// requested number of Nodes when executed with a fake client-set.
+func TestRunCreateNodesOp(t *testing.T) {
```

Comment:
I prefer to make this test more general and turn it into a test for `runOp()`. I'm guessing we should be able to test the resource-related `op`s (createNodeOps, createNamespacesOp, etc) in the same test.

Reply:
@sanposhiho Thanks for your review 🙏 It's worth discussing.
We can do it, but I don't like your idea one bit. The file will be too big, and there will be a little common test code among these ops.  Also, `scheduler_perf.go` is already too big for me to read easily. I think it should be split for readability in the future. Then it could be reasonable to separate each operation file. Considering this, I prefer not to become more general.

Also, test cases can be in separate files, as in QHint, but this will prevent VSCode or other editors from running individual test cases. This is a factor that makes debugging more difficult.

Reply:
> The file will be too big

We're going to use scheduler_perf_test.go either way even if we split the tests. Also, the file size doesn't much matter. Rather, having a similar testing code in different file's different tests is not great.

> there will be a little common test code among these ops.

For resource operation ops, what we need to do is simply (1) run the operation and (2) make sure the resource is modified. So, rather I think there'll be many common parts. The only part that have to be different is when you get the resources after `runOp()`.

> I think it should be split for readability in the future. Then it could be reasonable to separate each operation file.

I don't agree to have different files for different operations. There is no need.

Reply:
If this way is acceptable, no problem. I'll update it.
https://github.com/kubernetes/kubernetes/pull/131607#discussion_r2075333496